quantum/matrix.c

```diff
@@ -184,12 +194,12 @@ __attribute__((weak)) void matrix_read_cols_on_row(matrix_row_t current_matrix[]
         uint8_t pin_state = readMatrixPin(col_pins[col_index]);
 
         // Populate the matrix row with the state of the col pin
-        current_row_value |= pin_state ? 0 : row_shifter;
+        current_row_value |= pin_state ? PRESSED_KEY_PIN_STATE : row_shifter;
```

Comment:
Minimal fix (which still does not handle `NO_PIN` correctly — you need to fix `readMatrixPin()` for that):
```suggestion
        current_row_value |= (pin_state == PRESSED_KEY_PIN_STATE) ? 0 : row_shifter;
```
But if `readMatrixPin()` would always return 1 for an unpressed key and 0 for a pressed key, you won't need to change anything here.